### PR TITLE
tests: addition of test_example_app.py

### LINKED ISSUE #85: tests: addition of test_example_app.py
The package contains `examples/app.py` example application, but there is no test
for it in `tests/test_example_app.py`.  It should be added.  See existing examples:

- https://github.com/inveniosoftware/invenio-records/blob/master/tests/test_example_app.py
- https://github.com/inveniosoftware/invenio-marc21/blob/master/tests/test_example_app.py

### PR DESCRIPTION
**Pull request by hachreak**
* Adds test_example_app.py. (closes #85)

Signed-off-by: Leonardo Rossi <leonardo.r@cern.ch>

### CODE REVIEW

File: examples/app-recreate.sh [hunk 1]
```diff
@@ -0,0 +1,39 @@
+#!/bin/sh
+
+DIR=`dirname "$0"`
+
+# clean environment
+[ -e "$DIR/instance" ] && rm $DIR/instance -Rf
+# [ -e "$DIR/static" ] && rm $DIR/static -Rf
```
Comment (jirikuncar) [inline examples/app-recreate.sh]: If we don't need the command please remove it completely. Thanks

File: examples/app-recreate.sh [hunk 2]
```diff
@@ -0,0 +1,26 @@
+#!/bin/sh
```
Comment (jirikuncar) [inline examples/app-recreate.sh]: Can you split the file as you have done in other repos?

File: examples/app-recreate.sh [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+cd $DIR
+export FLASK_APP=app.py
+
+# clean environment
+[ -e "$DIR/instance" ] && rm $DIR/instance -Rf
+mkdir $DIR/instance
+
+# Install specific dependencies
+pip install -r requirements.txt || exit 1
```
Comment (jirikuncar) [inline examples/app-recreate.sh]: Use `set` instead.

File: examples/app-recreate.sh [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+flask db init || exit 1
+flask db create || exit 1
+
+# Create indices
+flask index destroy --yes-i-know --force
+flask index init || exit 1
+flask index queue init || exit 1
+
+# Load fixtures
+flask fixtures oaiserver || exit 1
```
Comment (jirikuncar) [inline examples/app-recreate.sh]: Add to `app-fixtures.sh`

File: tests/test_example_app.py [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+
+"""Test example app."""
+
+import os
+import signal
+import subprocess
+import time
+
+
+def setup_module(module):
```
Comment (jirikuncar) [inline tests/test_example_app.py]: Create `example_app` fixture instead. Thanks

File: examples/app-recreate.sh (hunk 3 above)
Comment (hachreak) [inline examples/app-recreate.sh]: done

File: examples/app-recreate.sh (hunk 4 above)
Comment (hachreak) [inline examples/app-recreate.sh]: done

File: tests/test_example_app.py (hunk 5 above)
Comment (hachreak) [inline tests/test_example_app.py]: done

**Comment by jirikuncar** [comment]
@hachreak LGTM (please rebase)

File: examples/app-recreate.sh (hunk 2 above)
Comment (hachreak) [inline examples/app-recreate.sh]: done

**Comment by hachreak** [comment]
rebased

**Comment by hachreak** [comment]
ping @jirikuncar

File: examples/app.py [hunk 6]
```diff
@@ -99,8 +75,6 @@
     OAISERVER_RECORD_INDEX='_all',
     OAISERVER_ID_PREFIX='oai:localhost:recid/',
     SECRET_KEY='CHANGE_ME',
-    SQLALCHEMY_DATABASE_URI=os.getenv('SQLALCHEMY_DATABASE_URI',
```
Comment (jirikuncar) [inline examples/app.py]: I have seen this in other PRs too. Please keep it so it's easy to change the database. Thanks

Comment (hachreak) [inline examples/app.py]: done

**Pull request merged**